### ISSUE TITLE
npm start

### ISSUE THREAD

**Issue by devsignCo** [NodeOS/NodeOS#157]
Hi Guys, I'm getting a strange issue starting NodeOs

$ npm start

> NodeOS@0.0.0 start /Users/path/to/nodeos/NodeOS
> scripts/start

fs.js:810
  return binding.readlink(pathModule._makeLong(path));
                 ^
Error: ENOENT, no such file or directory 'node_modules/nodeos-barebones/bzImage'
    at Error (native)
    at Object.fs.readlinkSync (fs.js:810:18)
    at Object.<anonymous> (/Users/path/to/nodeos/NodeOS/scripts/start:8:14)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

npm ERR! Darwin 14.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "start"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE
npm ERR! NodeOS@0.0.0 start: `scripts/start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the NodeOS@0.0.0 start script 'scripts/start'.
npm ERR! This is most likely a problem with the NodeOS package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     scripts/start
npm ERR! You can get their info via:
npm ERR!     npm owner ls NodeOS
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/christopher/Documents/nodeos/NodeOS/npm-debug.log

**Comment by piranna** [comment]
Have you executed ```npm install``` first? Also:

> npm ERR! Darwin 14.4.0

NodeOS can't be currently build on OSX, it needs a Linux host.

**Comment by devsignCo** [comment]
Oh right, let me check it out on my linux machine quickly.

**Comment by piranna** [comment]
Ok, I'll wait ;-)

**Comment by devsignCo** [comment]
okay so on my linux box I get the following during "npm install"

npm WARN package.json nodeos-barebones@0.0.0 No repository field.
npm WARN package.json nodeos-initramfs@0.0.0 No repository field.
npm WARN package.json nodeos-rootfs@0.0.0 No repository field.
npm WARN package.json nodeos-usersfs@0.0.0 No repository field.

> nodeos-barebones@0.0.0 preinstall /home/path/to/NodeOS/node_modules/nodeos-barebones
> scripts/preinstall

npm ERR! weird error 1
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! not ok code 0

**Comment by piranna** [comment]
> npm WARN This failure might be due to the use of legacy binary "node"
> npm WARN For further explanations, please read
> /usr/share/doc/nodejs/README.Debian

You need to install the package "nodejs" or "nodejs-legacy", not "node".

**Comment by devsignCo** [comment]
okay now i'm getting

Making info in doc
WARNING: `makeinfo' is missing on your system.  You should only need it if
         you modified a `.texi' or `.texinfo' file, or any other file
         indirectly affecting the aspect of the manual.  The spurious
         call might also be the consequence of using a buggy `make' (AIX,
         DU, IRIX).  You might want to install the `Texinfo' package or
         the `GNU make' package.  Grab either from any GNU archive site.
make[3]: *** [bfd.info] Error 1
Making info in po
config.status: creating po/Makefile.in
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing default-1 commands
config.status: executing bfd_stdint.h commands
config.status: executing default commands
make[2]: *** [info-recursive] Error 1
make[1]: *** [all-bfd] Error 2
make: *** [all] Error 2
Error compiling '/home/path/to/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/obj/nocona/binutils'
npm ERR! weird error 22
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! not ok code 0

**Comment by piranna** [comment]
I don't know why you got this error. Have you removed the "node" package? What do you get when you exec "node -v"? Have you installed the OS dependencies?

Why the way, what Linux distro are you using?

**Comment by devsignCo** [comment]
$ node -v
v0.10.25

I'm running Elementary OS

haha perhaps it's just not my time yet to try out NodeOS :D

**Comment by piranna** [comment]
> I'm running Elementary OS

It shouldn't give you so much problems since it's based on Ubuntu, that's our canonical one. Have you exec the ```bin/install-dependencies``` script?


> perhaps it's just not my time yet to try out NodeOS :D

NodeOS is currently in pre-alpha status, I've worked too much to make it easy to build but probably there are still some flaws... If you just want to give it a try, you can find pre-compiled builds ready to use in the releases tab.

**Comment by piranna** [comment]
@devsignCo, any update on this?

**Comment by tylerl0706** [comment]
Hi! I'd like to jump in and say I'm getting the same issues! I have run bin/install-dependencies too

I'm on ubuntu server 14.04LTS

**Comment by piranna** [comment]
14.04 had some dependencies issues regarding syslinux, that's the version available on SemaphoreCI. I don't remeber the commands, but I believe in /bin or somewhere else there's a script that install all the correct packages for 14.04. On higher versions there's no problem, but if more people arise with it, I was planning on remove the dependencies and download and compile syslinux as part of the generation process, so pull-requests are welcome! :-)

**Comment by tylerl0706** [comment]
Okay! So I'm back with Ubuntu server 15.04 and I followed the steps and did install-dependencies and I got this now:

In file included from /home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/function.h:25:0,
                 from insn-recog.c:10:
/home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/tree.h:3277:11: warning: identifier 'thread_local' conflicts with C++ keyword [-Wc++-compat]
  unsigned thread_local : 1;
           ^
gcc: internal compiler error: Killed (program cc1)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
Makefile:1070: recipe for target 'insn-recog.o' failed
make[1]: *** [insn-recog.o] Error 4
make[1]: *** Waiting for unfinished jobs....
Makefile:3842: recipe for target 'all-gcc' failed
make: *** [all-gcc] Error 2
Error compiling '/home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/obj/nocona/gcc'

npm ERR! nodeos-cross-toolchain@0.0.0 install: `scripts/install`
npm ERR! Exit status 32
npm ERR!
npm ERR! Failed at the nodeos-cross-toolchain@0.0.0 install script.
npm ERR! This is most likely a problem with the nodeos-cross-toolchain package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     scripts/install
npm ERR! You can get their info via:
npm ERR!     npm owner ls nodeos-cross-toolchain
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.19.0-22-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! cwd /home/azureuser/NodeOS
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian

npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /home/azureuser/NodeOS/npm-debug.log
npm ERR! not ok code 0

**Comment by piranna** [comment]
Gcc got killed, probably due to an out of memory error. Exec 'npm install --jobs=1', will be (a lot) slower but you'll don't get that error.

**Comment by tylerl0706** [comment]
Okay so I left my computer on to give that guy a try and I'm flooded with these after calling "sudo npm install --jobs=1" it hasn't crashed... but it's been like 8 hours!

In file included from /home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/function.h:25:0,
                 from insn-peep.c:15:
/home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/tree.h:3277:11: warning: identifier 'thread_local' conflicts with C++ keyword [-Wc++-compat]
  unsigned thread_local : 1;
           ^
In file included from insn-preds.c:9:0:
/home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/tree.h:3277:11: warning: identifier 'thread_local' conflicts with C++ keyword [-Wc++-compat]
  unsigned thread_local : 1;
           ^
In file included from /home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/function.h:25:0,
                 from insn-recog.c:10:
/home/azureuser/NodeOS/node_modules/nodeos-barebones/node_modules/nodeos-cross-toolchain/deps/gcc/gcc/tree.h:3277:11: warning: identifier 'thread_local' conflicts with C++ keyword [-Wc++-compat]
  unsigned thread_local : 1;

**Comment by piranna** [comment]
That output seems normal, I have it too. It has been 8 hours because you are not compiling in parallel because you have too few memory to do it, and I suspect that you are running your Linux instance in a virtual machine. If you use native hardware with parallel compilation, it shouldn't lack more that half an hour... On an old dual core laptop I compile everything from scratch in 75 minutes.

Did you be able to start it with 'npm start', or did the npm install failed?

**Comment by tylerl0706** [comment]
The npm install failed :( I am running it off of a server somewhere. So yeah, probably not native hardware. I gave it some more memory. I'll let you know how it goes

**Comment by piranna** [comment]
@tylerl0706, I suppose we can close this, isn't it?

**Comment by tylerl0706** [comment]
Yeah that sounds good. :)

**Comment by piranna** [comment]
Then let's close it :-) Maybe we'll review in the future how to build it on OSX, but that's another issue :-)
